base/src/main/java/com/smartdevicelink/managers/lifecycle/BaseSystemCapabilityManager.java

```diff
@@ -246,6 +255,7 @@ public WindowCapability getWindowCapability(int windowID) {
         for (WindowCapability windowCapability : display.getWindowCapabilities()) {
             int currentWindowID = windowCapability.getWindowID() != null ? windowCapability.getWindowID() : PredefinedWindows.DEFAULT_WINDOW.getValue();
             if (currentWindowID == windowID) {
+                windowCapability.setWindowID(windowID);
```

Comment:
Any particular reason that we're editing this value in a getter? It seems to me that this would basically act identically without this modification, and it's not great practice to modify internal values within a getter.

Reply:
I suppose if you need the caller to have this windowID in all cases, you could maybe make a copy of the capability object and modify that?

Reply:
I added a comment in the code, explaining, This is in alignment with iOS and JavaScript PR's for this issue that are linked in the comments on the trello card.

Reply:
Sorry, I suppose I was just saying that `DEFAULT_WINDOW` was logically equivalent to `null` in this case, based on the spec. 

I'm aware that this was part of the alignment with iOS. Unfortunately it's still bad practice. Realistically if we want this value to always be present, we should be ensuring this when the value is set instead. 

That said, here's a potential alternative option which doesn't involve too many changes, and doesn't modify the capabilities directly:
```suggestion
                if (windowCapability.getWindowID() == null) {
                    WindowCapability updatedCaps = new WindowCapability(windowCapability.getStore());
                    updatedCaps.setWindowID(PredefinedWindows.DEFAULT_WINDOW.getValue());
                    return updatedCaps;
                }
```

Reply:
Because developers can modify the stored window capability that they get through SystemCapabilityManager, I cloned them to prevent that.